pkg/apis/operator/v1alpha1/validation/garden.go

```diff
@@ -392,27 +392,33 @@ func validateGardenerAPIServerConfig(config *operatorv1alpha1.GardenerAPIServerC
 	}
 
 	if config.EncryptionConfig != nil {
-		seenResources := sets.New[string]()
+		var seenResources []schema.GroupResource
 
 		for i, resource := range config.EncryptionConfig.Resources {
-			idxPath := fldPath.Child("encryptionConfig", "resources").Index(i)
-			if seenResources.Has(resource) {
-				allErrs = append(allErrs, field.Duplicate(idxPath, resource))
+			var (
+				idxPath = fldPath.Child("encryptionConfig", "resources").Index(i)
+				gr      = schema.ParseGroupResource(resource)
+			)
+
+			if strings.ToLower(resource) != resource {
+				allErrs = append(allErrs, field.Invalid(idxPath, resource, "resource must be lower case"))
 			}
 
-			if gardenerutils.DefaultGardenerResourcesForEncryption().Has(resource) {
+			if slices.Contains(seenResources, gr) {
+				allErrs = append(allErrs, field.Duplicate(idxPath, resource))
+			}
+			if slices.Contains(gardenerutils.DefaultGardenerGRsForEncryption(), gr) {
 				allErrs = append(allErrs, field.Forbidden(idxPath, fmt.Sprintf("%q are always encrypted", resource)))
 			}
-
-			if !operator.IsServedByGardenerAPIServer(resource) {
-				allErrs = append(allErrs, field.Invalid(idxPath, resource, "should be a resource served by gardener-apiserver. ie; should have any of the suffixes {authentication,core,operations,security,settings,seedmanagement}.gardener.cloud"))
+			if gr.Resource == "*" || gr.Group == "*" {
```

Comment:
Can we do this? It seems simpler.

```suggestion
			if strings.Contains(resource, "*") {
```